foreign-types-shared/src/lib.rs

```diff
@@ -25,6 +25,9 @@ pub trait ForeignType: Sized {
 
     /// Returns a raw pointer to the wrapped value.
     fn as_ptr(&self) -> *mut Self::CType;
+
+    /// Consumes the wrapper and returnes the raw pointer
+    fn into_ptr(self) -> *mut Self::CType;
```

Comment:
We should be able to implement this here rather than in the macro, which will avoid breaking back compat as well.

Reply:
sure, I think we could extract the `into_ptr` to a standalone `IntoForeignType` trait.

Reply:
It doesn't need to be a separate trait. You can just define the new method as

```rust
fn into_ptr(self) -> *mut Self::CType {
    let ptr = self.as_ptr();
    mem::forget(self);
    ptr
}
```

Reply:
sorry, it seems I misunderstand your comment, thanks